README.md

```diff
@@ -1,19 +1,311 @@
+```
+ sbt run -Dapplication.router=testOnlyDoNotUseInAppConf.Routes
+```
+
+## Endpoints
+
+### `POST /hec/tax-check`
+Generate a tax check code and save the tax check data. A `201` (CREATED) response is 
+returned if the tax check has successfully been saved. Requires an active GG or Stride session.
+
+Example request body:
+```json
+{
+   "applicantDetails": {
+      "ggCredId": "credId",
+      "name": { 
+         "firstName": "First",
+         "lastName": "Last"
+      },
+      "dateOfBirth":"19221201"
+   },
+   "licenceDetails":{
+      "licenceType": "DriverOfTaxisAndPrivateHires",
+      "licenceTimeTrading": "ZeroToTwoYears",
+      "licenceValidityPeriod": "UpToThreeYears"
+   },
+   "taxDetails": { 
+      "nino": "AB123456C",
+      "sautr": "1234567890",
+      "taxSituation":"SA",
+      "saIncomeDeclared": "Yes",
+      "saStatusResponse": {
+         "sautr": "1234567890",
+         "taxYear": 2020,
+         "status": "ReturnFound"
+      },
+      "relevantIncomeTaxYear": 2020,
+      "correctiveAction": "DormantAccountReactivated"
+   },
+   "taxCheckStartDateTime": "2021-12-02T14:56:11.75Z[Europe/London]",
+   "source":"Stride",
+   "type":"Individual"
+}
+```
+Example response body:
+```json
+{
+   "taxCheckData" : {
+      "applicantDetails": {
+         "ggCredId": "credId",
+         "crn": "1234567890",
+         "companyName": "Test Company"
+      },
+      "licenceDetails": {
+         "licenceType": "OperatorOfPrivateHireVehicles",
+         "licenceTimeTrading": "TwoToFourYears",
+         "licenceValidityPeriod": "UpToOneYear"
+      },
+      "taxDetails": {
+         "desCTUTR" :"1111111111",
+         "userSuppliedCTUTR": "1111111111",
+         "ctIncomeDeclared": "No", 
+         "ctStatus": {
+            "ctutr":  "1234567890",
+            "startDate":  "20201001",
+            "endDate":  "20193009",
+            "latestAccountingPeriod": {
+               "startDate": "20200131",
+               "endDate": "20200524",
+               "ctStatus": "NoticeToFileIssued"
+            }
+         }, 
+         "recentlyStaredTrading": "No",
+         "chargeableForCT": "Yes",
+         "correctiveAction": "Other"
+      },
+      "taxCheckStartDateTime": "2021-12-02T14:56:11.75Z[Europe/London]",
+      "source": "Digital",
+      "type": "Company"
+   },
+   "taxCheckCode": "TLAR4D6HB",
+   "expiresAfter": "2022-04-01",
+   "createDate": "2021-12-02T14:56:49.618Z[Europe/London]",
+   "isExtracted": false
+}
+```
+N.B. the combinations of answers in the request and response bodies above do not correspond to actual
+real business scenarios. A value for each field (even if optional) has been provided for illustrative 
+purposes.
+
+
+### `POST /hec/match-tax-check`
+Check the details given in the request to see if they match an existing tax check in the database. A
+`200` (OK) response if returned if a match result can be given.
+
+Example request body:
```

Comment:
can we mention that this POST body is to check the tax check code for Individual and below one is for company?

Reply:
updated [here](https://github.com/hmrc/hec/pull/53/commits/029cc5817c153a21799cc847141b6f5f18e2e6c3)